### ISSUE TITLE
feature: rebase to Kubernetes 1.32.3

### ISSUE THREAD

**Issue by embik** [kcp-dev/kcp#3209]
### Feature Description

Kubernetes 1.32.0 is slated to come out [tomorrow](https://www.kubernetes.dev/resources/release/#summary). We should rebase kcp to that new version. A final changelog isn't out yet, so it likely makes sense to wait for that to understand what changes to expect from the rebase.

### Proposed Solution

Follow [the rebase guide](https://docs.kcp.io/kcp/main/contributing/rebasing-kubernetes/).

### Want to contribute?

- [ ] I would like to work on this issue.

**Comment by gman0** [comment]
Taking (I would self-`/assign` but that probably won't work for me at the moment because of perms).

**Comment by gman0** [comment]
WIP branches for [kcp-dev/kubernetes](https://github.com/kcp-dev/kubernetes):
* https://github.com/gman0/kubernetes/tree/kcp-1.32-baseline
* https://github.com/gman0/kubernetes/tree/kcp-1.32-pre.1-deps
* https://github.com/gman0/kubernetes/tree/kcp-1.32-pre.2-fixes

**Comment by gman0** [comment]
WIP branch for [kcp-dev/kcp](https://github.com/kcp-dev/kcp):
* https://github.com/gman0/kcp/tree/1.32-bump
* https://github.com/gman0/kcp/tree/1.32-bump-pre.1

Also working on updating the rebasing docs:
* https://github.com/gman0/kcp/tree/update-rebase-docs

**Comment by gman0** [comment]
https://github.com/gman0/kubernetes/commit/3ecc3a49cd2572a1bc9404d5d1aed30e8bf25fc6

TL;DR we may need to wire in context to the `asPartialObjectMetadataList` function. Will try it out and we can discuss it once the PR is out.

**Comment by gman0** [comment]
https://github.com/gman0/kcp-code-generator/commit/e65054c73f6315664236041ef63e850fd217a808

The build constraint `!ignore_autogenerated` in templates caused the generated files to be not picked up by tools in https://github.com/kubernetes/code-generator/tree/master/cmd, specifically when re-generating deepcopy functions as a part of [`hack/update-codegen.sh`](https://github.com/kubernetes/kubernetes/blob/master/hack/update-codegen.sh).

Removing these constraints fixed the issue.

TODO:
* It's not yet clear why with this worked, _with the constraints_, when publishing our k8s rebase to v1.31.0. Generating the same v1.31.0 code now **fails** however (could this be some external dependency that was updated?).
* The build constraint seems to have been introduced already as a part of first PR: https://github.com/kcp-dev/code-generator/pull/1/files#diff-8f8bddc9a9b92d14d0cbdd05deaace70831ec9de0879b4b5b1bd99bc5f41a81eR21-R22 . I still need to check more thoroughly why it was there in the first place.

**Comment by gman0** [comment]
Hi folks, reviving this issue with the latest updates:

* [[DIFF-ONLY] Rebase to 1.32.3 kcp-dev/apimachinery#84](https://github.com/kcp-dev/apimachinery/pull/84)
* [[DIFF-ONLY] Rebase to 1.32.3 kcp-dev/code-generator#105](https://github.com/kcp-dev/code-generator/pull/105)
* [[DIFF-ONLY] Rebase to 1.32.3 kcp-dev/client-go#41](https://github.com/kcp-dev/client-go/pull/41)
* [[DIFF-ONLY] Rebase to 1.32.3 kcp-dev/kubernetes#166](https://github.com/kcp-dev/kubernetes/pull/166)
* kcp: https://github.com/kcp-dev/kcp/pull/3383

There were only minor changes compared to https://github.com/kcp-dev/kubernetes/pull/161
